gateway/src/main/java/com/hurence/webapiservice/http/ingestion/ImportRequestParser.java

```diff
@@ -14,61 +16,49 @@
     private static final Logger LOGGER = LoggerFactory.getLogger(ImportRequestParser.class);
 
 
-    public String[] checkRequest (JsonArray metricsParam) {
-        final String[] error = new String[1];
-        try {
-            if (null == metricsParam) {
-                error[0] = "parameter array is NULL";
-                throw new IllegalArgumentException("parameter array is NULL");
-            }else if (metricsParam.isEmpty()) {
-                error[0] = "parameter array is empty";
-                throw new IllegalArgumentException("parameter array is empty");
+    public ResponseAndErrorHolder checkAndBuildValidHistorianImportRequest(JsonArray metricsParam) throws IllegalArgumentException {
+        // TODO should i keep this ?
+        /*if (null == metricsParam) {
+            throw new IllegalArgumentException("parameter array is NULL");
+        }else if (metricsParam.isEmpty()) {
+            throw new IllegalArgumentException("parameter array is empty");
+        }*/
+
+        ResponseAndErrorHolder responseAndErrorHolder = new ResponseAndErrorHolder();
+        for (Object object : metricsParam) {
+            JsonObject timeserie = (JsonObject) object;
+            if ((timeserie.getString(METRIC_NAME_REQUEST_FIELD) == null) || (timeserie.getJsonArray(POINTS_REQUEST_FIELD) == null)) {
+                responseAndErrorHolder.errorMessages.add("can't add this object " + timeserie.toString());
```

Comment:
Once again I disagree with you error message. "the timeserie number 3 (place in the array) is not well formed" is a better information for the client so he can locate the problem.